app/assets/javascripts/app/form-validation.js

```diff
@@ -1,16 +1,84 @@
 import 'classlist.js';
-import gf from 'gentleform';
+import h5f from 'h5f';
 
 
-function validateForm() {
-  const form = document.querySelector('form[novalidate]');
+const validate = {
+  msgs: {
+    missing: 'Please fill in all required fields.',
+    mismatch: 'Please match the requested format.',
+  },
 
-  if (form) {
-    gf(form, function onSubmit(e) {
-      if (!this.isValid()) e.preventDefault();
+  init() {
+    this.form = document.querySelector('form');
+    if (!this.form) return;
+    this.btn = this.form.querySelector('[type=submit]');
+
+    h5f.setup(this.form, {
+      validClass: 'valid',
+      invalidClass: 'invalid',
+      requiredClass: 'required',
+      placeholderClass: 'placeholder',
     });
-  }
-}
+
+    this.addEvents();
+  },
+
+  addEvents() {
+    this.form.addEventListener('invalid', e => e.preventDefault(), true);
+    this.form.addEventListener('change', e => this.validateField(e.target));
+    this.form.addEventListener('submit', e => this.submitForm(e));
+    this.btn.addEventListener('click', () => this.validateForm());
+  },
+
+  submitForm(e) {
+    if (!this.form.checkValidity()) e.preventDefault();
+  },
+
+  validateForm() {
+    const fields = this.form.querySelectorAll('.field');
+    for (let i = 0; i < fields.length; i++) {
+      this.validateField(fields[i]);
+    }
+
+    // add focus to first invalid input
+    const invalidField = this.form.querySelector(':invalid');
+    if (invalidField) {
+      invalidField.focus();
+    }
+  },
+
+  validateField(f) {
+    const parent = f.parentNode;
+    const errorMsg = parent.querySelector('.error-message');
+
+    if (errorMsg !== null) parent.removeChild(errorMsg);
+    f.classList.add('interacted');
```

Comment:
would move this to first line in method:

`
f.classList.add('interacted');

const parent = f.parentNode;
...
`